### PR TITLE
bubble sort programs

### PR DESCRIPTION
**Pull request by Ahmed-Benj**
Hello TSE,

Please review my programs which do :

bubble sort program using two methods ==

1) bubble_sort : classic method that takes numbers array as argument
2) bubble_sort_by : method using yield method that takes string array as argument

Input : unsorted array
Output : sorted array

Kind regards

### CODE REVIEW

**Review by Srjnnnn — CHANGES_REQUESTED** [review]
## Status: Changes Required :recycle: 
Hello @Ahmed-Benj. I cloned your repo and move the feature-branch and I just found one XML file nothing more. Also, your linters didn't pass. I believe there is confusion about what you have submitted, please take a look and give us some ruby code to review. :confused: 

You need to arrange bubble_sort and bubble_sort_by methods. You can find the necessary details [here](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/57421588/).

If there is a question about the review, feel free to reach me via Slack. :100: 
_Eyüp Sercan UYGUR_

**Comment by Ahmed-Benj** [comment]
There was a mistake in git pushing, I am sorry for the inconvenience, I have fixed that.

I am working on linters

**Comment by descholar-ceo** [comment]
# This Pull Request is under review by [descholar-ceo](https://github.com/descholar-ceo) :eyes: :eyes: 
## Please wait a bit ...

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        substitute_value = array[index]
+        array[index] = array[index + 1]
+        array[index + 1] = substitute_value
+      end
+      index += 1
+    end
+    number_of_times_sort += 1
+    index = 0
+  end
+  print array
```
Comment (descholar-ceo) [inline bubble_sort.rb:16]: use return statement, not print

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+end
+
+bubble_sort([4, 3, 78, 2, 0, 2])
```
Comment (descholar-ceo) [inline bubble_sort.rb]: remove this line of code => this is a debugging statement

File: bubble_sort_by.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        substitute_value = left
+        left = right
+        right = substitute_value
+      end
+      index += 1
+    end
+    number_of_times_sort += 1
+    index = 0
+  end
+  print array
```
Comment (descholar-ceo) [inline bubble_sort_by.rb:17]: use return instead of print

File: bubble_sort_by.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+end
+
+bubble_sort_b(%w[hi hello hey])
```
Comment (descholar-ceo) [inline bubble_sort_by.rb]: remove this line of code as well => it's a debugging statement

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
# Status: Changes Required :vertical_traffic_light: :warning: 
## Hello @Ahmed-Benj  :handshake: :handshake: :handshake: 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: I liked that you have tried to implemented the requested changes from the previous review, keep it up, and your project is really almost done.
#### :warning: However we need to make few changes to mark this project approved, don't feel discouraged because of being asked some changes more and more, that's beacuse this is a learning process, and we are here to help you to become an out-standing developer. See the requested change below :point_down: :point_down: :point_down: 
### Note: All of requested changes which are marked as [Required] are mandatory to be implemented in order to get this project approved, others which are marked as [Optional] are nice to have in your project to make it outstanding

# [Required Changes] :red_circle: 
- [x] This branch has conflicts with the base branch, please fix conflicts before you submit for another review
- [x] Work again on this Pull Request description, currently the description is not reflecting the work you have accomplished on this PR, please improve it by answering this questions, what does this PR do? which tasks need to be completed on this PR?
- [x] Remove .idea folder from root driectory of your project => that folder is not related to your code
- [x] Add a custom readme file => follow [microverse's readme template](https://github.com/microverseinc/readme-template)
- [x] bubble_sort _by should be a method, not a block. A method has this format:
```ruby
    def method_name(params)
        # some codes here
        # return statement
    end
```
- [x] Use return statement instead of printing a result
- [x] Remove all of the debugging statements
- [x] Implement all changes I suggested in inline comments in your codes

# [Optional changes] :yellow_circle: 
:warning: Your commit messages should be in imperative mode or present tense, _i.e.:_ you should have written `Add bubble sort method to bubble_sort.rb file` instead of writing `bubble sort programs`, refer [to this article for more details on how to write a good && professional commit message](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8)

___
_If anything seems to be unclear to you, reachout and ask anything by commenting on this Pull Request and mention me as [descholar-ceo](https://github.com/descholar-ceo), I will receive a notification and will respond._

[GitHub](https://github.com/descholar-ceo) | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

_Reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)_

File: bubble_sort_by.rb (hunk 4 above)
Comment (Ahmed-Benj) [reply bubble_sort_by.rb]: fixed

File: bubble_sort_by.rb (hunk 3 above)
Comment (Ahmed-Benj) [reply bubble_sort_by.rb:17]: fixed

File: bubble_sort.rb (hunk 2 above)
Comment (Ahmed-Benj) [reply bubble_sort.rb]: fixed

File: bubble_sort.rb (hunk 1 above)
Comment (Ahmed-Benj) [reply bubble_sort.rb:16]: fixed

**Comment by Ahmed-Benj** [comment]
Hello,

thank you for reviewing my project, this project is sampling basic sorting program using ruby.

We have used two methods to do the sort :

classic method of calling array
yielding arguments constituting array pairs

Kind regards,

**Review by layzrd — CHANGES_REQUESTED** [review]
Hi @Ahmed-Benj, I am Abdulaziz Redi Ali, your TSE in this PR. 👏 You did very well on this project. ✨ I applaud you for working hard on implementing the bubble sort algorithm. Most requirements have done 👌, Kindly fix the following issues.

[REQUIRED]

Fix the Rubocop linting errors
Update/edit the pull request message it should be the tasks you have accomplished 

[OPTIONAL]

Update the while loop to stop iterating if the array is sorted before reaching the last element. It will improve code efficiency.

Once you fix the issues, kindly submit another code review. ♲

 Happy coding 👨‍💻
 Abdulaziz Redi Ali

**Review by arslanbisharat — APPROVED** [review]
# Approved :smile: 

Hello Developer :wave:

I just reviewed your project and I am very happy to inform you that your project is approved.
Congratulations on achieving this milestone in your career. :clap:


_I am looking forward to seeing your next project._

Stay safe and give me a smile 😇

### Happy Coding
------------------------------------------------------------------------------------------------
**Regards,
Arslan Bisharat
slack: arslanbisharat**

**Pull request merged**